metricbeat/helper/elastic/elastic.go

```diff
@@ -158,8 +158,31 @@ func NewModule(base *mb.BaseModule, xpackEnabledMetricsets []string, logger *log
 		return nil, errors.Wrapf(err, "could not unpack configuration for module %v", moduleName)
 	}
 
-	// These metricsets are exactly the ones required if xpack.enabled == true
-	raw["metricsets"] = xpackEnabledMetricsets
+	// Ensure all required metricsets are enabled when xpack.enabled == true
+	cfgdMetricsets, err := raw.GetValue("metricsets")
+	if err != nil || cfgdMetricsets == nil || len(cfgdMetricsets.([]interface{})) == 0 {
```

Comment:
The code here and below does somehow look too complicate for what it does but currently not deep enough into the code to recommend adjustements.

Reply:
This is included in the blocking PR, let's discuss it here: https://github.com/elastic/beats/pull/34273/files#r1081193171